hist/hist/inc/TH1.h

```diff
@@ -137,6 +215,143 @@ class TH1 : public TNamed, public TAttLine, public TAttFill, public TAttMarker {
    TH1(const TH1&) = delete;
    TH1& operator=(const TH1&) = delete;
 
+   /**
+    * @brief Slices a histogram in place based on the specified bin ranges for each dimension.
+    * 
+    * This function modifies the histogram by extracting a sub-region defined by the provided 
+    * bin ranges for each dimension. The resulting histogram will have updated bin counts, 
+    * edges, and contents. Bin contents outside the range fall into the flow bins.
+    * This function is used by the python implementation of the Unified Histogram Interface (UHI)
+    * for slicing.
+    * 
+    * @tparam ValueType The type of the histogram's data.
+    * @param args A vector of integers specifying the low and upper edges of the slice
+    *             for each dimension.
+    * @param fArray A pointer to the histogram's data array.
+    * @param fN The size of fArray.
+    */
+   template <typename ValueType>
+   void SliceHistoInPlace(std::vector<Int_t>& args, ValueType*& fArray, Int_t& fN) {
+      constexpr Int_t kMaxDim = 3;
+      const auto ndim = args.size() / 2;
+      if (ndim != static_cast<size_t>(fDimension)) {
+         throw std::invalid_argument("Number of dimensions in slice does not match histogram dimension.");
+      }
+
+      // Compute new bin counts and edges
+      std::array<Int_t, kMaxDim> nBins{}, totalBins{};
+      std::array<Double_t, kMaxDim> lowEdge{}, upEdge{};
+      for (size_t d = 0; d < ndim; ++d) {
+         const auto &axis = (d == 0 ? fXaxis : d == 1 ? fYaxis : fZaxis);
+         auto start = std::clamp(args[2*d], 1, axis.GetNbins());
+         auto end   = std::clamp(args[2*d+1], 1, axis.GetNbins()+1);
+         nBins[d] = end - start;
+         lowEdge[d] = axis.GetBinLowEdge(start);
+         upEdge[d] = axis.GetBinLowEdge(end);
+         totalBins[d] = axis.GetNbins() + 2;
+         args[2*d] = start; args[2*d+1] = end;
+      }
+
+      // Compute layout sizes for slice
+      size_t rowSz  = nBins[0] + 2;
+      size_t planeSz= rowSz * (ndim>1 ? nBins[1]+2 : 1);
+      size_t newSize= planeSz * (ndim>2 ? nBins[2]+2 : 1);
+
+      // Allocate the new array
+      auto *newArr = new ValueType[newSize]();
+
+      auto rowIncr = 1 + (ndim > 1 ? (rowSz-1) : 0);
+      ValueType under=0, over=0;
+      Bool_t firstUnder = false;
+      Int_t lastOverIdx = 0;
+
+      // Copy the valid slice bins
+      size_t dstIdx = 1 + (ndim > 1 ? rowSz : 0) + (ndim > 2 ? planeSz : 0);
+      for (auto z = (ndim>2?args[4]:0); z < (ndim>2?args[5]:1); ++z) {
+         for (auto y = (ndim>1?args[2]:0); y < (ndim>1?args[3]:1); ++y) {
+               size_t rowStart = z*totalBins[1]*totalBins[0] + y*totalBins[0] + args[0];
+               std::copy_n(fArray+rowStart, nBins[0], newArr+dstIdx);
+               if (!firstUnder) {
+                  under = std::accumulate(fArray, fArray+rowStart, 0.0);
```

Comment:
To be perfectly consistent
```suggestion
                  under = std::accumulate(fArray, fArray+rowStart, ValueType{});
```

and same comment applies to other `std::accumulate` calls